src/main/java/com/cloudbees/jenkins/support/util/WordReplacer.java

```diff
@@ -0,0 +1,168 @@
+package com.cloudbees.jenkins.support.util;
+
+import java.util.Locale;
+
+public class WordReplacer {
+    /**
+     * Replace all words in the input by the replaces. The replacements happens only if the texts to replace are not
+     * part of a greater word, that is, if the text is a whole word, separated by non {@link Character#isLetterOrDigit(char)}
+     * characters.
+     * @param input the text where the replacements take place
+     * @param words the words to look for and replace
+     * @param replaces the new words to use
+     */
+    public static String replaceWords(String input, String[] words, String[] replaces) {
+        StringBuilder sb = new StringBuilder(input);
+        replaceWords(sb, words, replaces);
+        return sb.toString();
+    }
+
+    /**
+     * Replace all words in the input by the replaces. The replacements happens only if the texts to replace are not
+     * part of a greater word, that is, if the text is a whole word, separated by non {@link Character#isLetterOrDigit(char)}
+     * characters.
+     * @param input the text where the replacements take place
+     * @param words the words to look for and replace
+     * @param replaces the new words to use
+     */
+    public static String replaceWordsIgnoreCase(String input, String[] words, String[] replaces) {
+        StringBuilder sb = new StringBuilder(input);
+        replaceWordsIgnoreCase(sb, words, replaces);
+        return sb.toString();
+    }
+
+    /**
+     * See {@link #replaceWords(String, String[], String[])}
+     * @param input the text where the replacements take place
+     * @param words the words to look for and replace
+     * @param replaces the new words to use
+     */
+    public static void replaceWords(StringBuilder input, String[] words, String[] replaces) {
+        replaceWordsIgnoreCase(input, words, replaces, false);
+    }
+
+    /**
+     * See {@link #replaceWords(String, String[], String[])}
+     * @param input the text where the replacements take place
+     * @param words the words to look for and replace
+     * @param replaces the new words to use
+     */
+    public static void replaceWordsIgnoreCase(StringBuilder input, String[] words, String[] replaces) {
+        replaceWordsIgnoreCase(input, words, replaces, true);
+    }
+
+    // To avoid repeat this code above
+    private static void replaceWordsIgnoreCase(StringBuilder input, String[] words, String[] replaces, boolean ignoreCase) {
```

Comment:
The "ignoreCase" parameter is never used?

Reply:
Right, I'm fixed some bugs in the class. Some methods are not used, I created them just in case I need them. I've also added a complete test to check all methods.